internal/api/v1alpha1/authproxyworkload_webhook.go

```diff
@@ -121,9 +121,33 @@ func (r *AuthProxyWorkload) validateUpdateFrom(op *AuthProxyWorkload) field.Erro
 			"selector cannot be changed on update"))
 	}
 
+	allErrs = append(allErrs, validateRolloutStrategyChange(r.Spec.AuthProxyContainer, op.Spec.AuthProxyContainer)...)
+
 	return allErrs
 }
 
+func validateRolloutStrategyChange(c *AuthProxyContainerSpec, oc *AuthProxyContainerSpec) []*field.Error {
```

Comment:
Let's add a doc string here explaining why this is important and what validate is doing.

Reply:
Fixed.